packages/components/src/components/Table/Table.test.js

```diff
@@ -0,0 +1,534 @@
+  ).toBeNull();
+  expect(queryByText('No Resources created under any namespace.')).toBeNull();
+
+  fireEvent.click(queryByLabelText('Select row'));
+  fireEvent.click(queryByText(/Delete/i));
+
+  expect(handleDelete).toHaveBeenCalledTimes(1);
+
+  expect(
+    handleDelete.mock.calls[0][0].map(row => {
+      return row.id;
+    })
+  ).toEqual(
+    props.rows.map(row => {
+      return row.id;
+    })
+  );
+
+  expect(typeof handleDelete.mock.calls[0][1]).toEqual('function');
+});
+
+it("Table's handle delete specifies correct arguments for callback with two rows selected", () => {
+  const handleDelete = jest.fn();
+
+  const props = {
+    batchActionButtons: batchActionButtons.slice(0, 1),
+    loading: false,
+    rows,
+    headers,
+    selectedNamespace: '*'
+  };
+  props.batchActionButtons[0].handler = handleDelete;
+  const { queryByText, queryByLabelText } = render(<Table {...props} />);
+
+  expect(queryByText(/Resources/i)).toBeNull();
+  expect(queryByText(/Name/i)).toBeTruthy();
+  expect(queryByText(/Namespace/i)).toBeTruthy();
+  expect(queryByText(/Date Modified/i)).toBeTruthy();
+  expect(queryByText(/resource-one/i)).toBeTruthy();
+  expect(queryByText(/namespace1/i)).toBeTruthy();
+  expect(queryByText(/16 minutes ago/i)).toBeTruthy();
+  expect(queryByText(/resource-two/i)).toBeTruthy();
+  expect(queryByText(/namespace2/i)).toBeTruthy();
+  expect(queryByText(/20 hours ago/i)).toBeTruthy();
+  expect(queryByText(/Delete/i)).toBeTruthy();
+  expect(queryByText(/Add/i)).toBeNull();
+  expect(queryByLabelText('Select all rows')).toBeTruthy();
+  expect(queryByLabelText('Select row')).toBeTruthy();
+  expect(
+    queryByText("No Resources created under namespace 'default'.")
+  ).toBeNull();
+  expect(queryByText('No Resources created under any namespace.')).toBeNull();
+
+  fireEvent.click(queryByLabelText('Select all rows'));
+  fireEvent.click(queryByText(/Delete/i));
+
+  expect(handleDelete).toHaveBeenCalledTimes(1);
+
+  expect(
+    handleDelete.mock.calls[0][0].map(row => {
+      return row.id;
+    })
+  ).toEqual(
+    props.rows.map(row => {
+      return row.id;
+    })
+  );
+
+  expect(typeof handleDelete.mock.calls[0][1]).toEqual('function');
+});
+
+it("Table's multiple buttons respond correctly", () => {
+  const handleDelete = jest.fn();
+  const handleAdd = jest.fn();
+  const handleRebuild = jest.fn();
+  const handleRebuildAll = jest.fn();
+
+  const props = {
+    batchActionButtons,
+    loading: false,
+    rows,
+    headers,
+    selectedNamespace: '*',
+    toolbarButtons
+  };
+  props.batchActionButtons[0].handler = handleDelete;
+  props.batchActionButtons[1].handler = handleRebuild;
+  props.toolbarButtons[0].handler = handleAdd;
+  props.toolbarButtons[1].handler = handleRebuildAll;
+  const { queryByText, queryByLabelText } = render(<Table {...props} />);
+
+  expect(queryByText(/Resources/i)).toBeNull();
+  expect(queryByText(/Name/i)).toBeTruthy();
```

Comment:
Thorough tests are great but generally there's no need to re-assert all of the basic content / behaviour on every test unless there's a chance the specific combination of props could break it. Tests should be relatively focused, and it should be easy to tell at a glance what the important content / behaviour is without having to figure out which pieces are essentially noise and should be ignored.